include/umf/providers/provider_cuda.h

```diff
@@ -34,13 +34,17 @@ umf_result_t umfCUDAMemoryProviderParamsDestroy(
 /// @brief  Set the CUDA context handle in the parameters struct.
 /// @param  hParams handle to the parameters of the CUDA Memory Provider.
 /// @param  hContext handle to the CUDA context.
+/// @note   This function is optional - if no context is set in the params
```

Comment:
`This function is optional` sounds confusing. I understand what you meant, but others might be confused.
Maybe we should add a note the `umfCUDAMemoryProviderParamsCreate` function and say that returned params structure is initialized with default value, and default context is a current context and the same for device.

Reply:
is it better now?